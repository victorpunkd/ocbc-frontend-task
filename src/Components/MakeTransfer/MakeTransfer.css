.makeTransferContainer {
  width: 100%;
}

.transferForm {
  height: 75vh;
  overflow: scroll;
  padding: 5% 0%;
  position: relative;
}

.transferForm::-webkit-scrollbar {
  display: none;
}

.formCenter {
  margin: 0;
  position: relative;
  top: 50%;
  transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  text-align: center;
}

.formElement {
  margin-top: 5%;
}

.footer {
  display: flex;
}

.buttonHolder {
  flex: 1;
  padding: 0% 2%;
}
