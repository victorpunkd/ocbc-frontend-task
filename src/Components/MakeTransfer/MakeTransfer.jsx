import React, { useState, useEffect, useCallback } from "react";
import { useNavigate } from "react-router-dom";
import "./MakeTransfer.css";
import NavBar from "../NavBar/NavBar";
import CustomButton from "../../CommonComponents/CustomButton/CustomButton";
import CustomSecondaryButton from "../../CommonComponents/CustomSecondaryButton/CustomSecondaryButton";
import CustomSnackBar from "../../CommonComponents/CustomSnackBar/CustomSnackBar";
import CustomBackdrop from "../../CommonComponents/CustomBackdrop/CustomBackdrop";

import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import TextField from "@mui/material/TextField";
import DesktopDatePicker from "@mui/lab/DesktopDatePicker";
import LocalizationProvider from "@mui/lab/LocalizationProvider";
import AdapterDateFns from "@mui/lab/AdapterDateFns";
import PublishIcon from "@mui/icons-material/Publish";
import CloseIcon from "@mui/icons-material/Close";
import axios from "axios";

const MakeTransfer = () => {
  const navigate = useNavigate();
  const [recipient, setRecipient] = useState([]);
  const [selectedRecipient, setSelectedRecipient] = useState("");
  const [transactionDate, setTransactionDate] = useState(new Date());
  const [description, setDescription] = useState("");
  const [amount, setAmount] = useState("");
  const [errorState, setErrorState] = useState({});
  const [isSnackbarOpen, setIsSnackbarOpen] = useState(false);
  const [snackBarSeverity, setSnackBarSeverity] = useState("");
  const [snackBarMessage, setSnackBarMessage] = useState("");
  const [isBackdropOpen, setIsBackdropOpen] = useState(false);

  const handleBackEndError = useCallback(
    (error) => {
      setIsSnackbarOpen(true);
      setSnackBarMessage(error?.response?.data?.description || "");
      setSnackBarSeverity("error");
      sessionStorage.removeItem("token");
      setTimeout(() => {
        navigate("/");
      }, 1000);
    },
    [navigate]
  );

  const getRecipients = useCallback(() => {
    setIsBackdropOpen(true);
    axios
      .get(`${process.env.REACT_APP_BASE_URL}account/payees`, {
        headers: {
          Authorization: sessionStorage.getItem("token"),
        },
      })
      .then((response) => {
        if (response.data && response.data.data)
          setRecipient(response.data.data);
      })
      .catch((error) => {
        if (error.response.data && error.response.data.status) {
          handleBackEndError(error);
        }
      })
      .finally(() => {
        setIsBackdropOpen(false);
      });
  }, [handleBackEndError]);

  useEffect(() => {
    if (!sessionStorage.getItem("token")) {
      navigate("/no-access");
    } else {
      getRecipients();
    }
  }, [navigate, getRecipients]);

  const handleRecipientChange = (event) => {
    setSelectedRecipient(event.target.value);
  };

  const handleDateChange = (newValue) => {
    let todaysDate = new Date();
    if (todaysDate.setHours(0, 0, 0, 0) <= newValue.setHours(0, 0, 0, 0)) {
      setTransactionDate(newValue);
    } else {
      setIsSnackbarOpen(true);
      setSnackBarSeverity("warning");
      setSnackBarMessage("Transaction date cant be less than todays date");
    }
  };

  const handleTextBoxChange = (type, event) => {
    if (type === "description") {
      setDescription(event.target.value);
    } else if (type === "amount") {
      if (!event.target.value.match(/[%<>\\$'"]/)) {
        setAmount(event.target.value); // only set when successful
      }
    }
  };

  const handleCancelClick = () => {
    navigate("/dashboard");
  };

  const validateDetails = () => {
    const errors = {};
    if (!selectedRecipient) {
      errors["selectedRecipient"] = "Required";
    }
    if (!transactionDate) {
      errors["transactionDate"] = "Required";
    }
    if (!description) {
      errors["description"] = "Required";
    }
    if (!amount) {
      errors["amount"] = "Required";
    }
    return errors;
  };

  const makeTransfer = () => {
    axios
      .post(
        `${process.env.REACT_APP_BASE_URL}transfer`,
        {
          recipientAccountNo: selectedRecipient,
          amount: amount,
          date: transactionDate,
          description: description,
        },
        {
          headers: {
            Authorization: sessionStorage.getItem("token"),
          },
        }
      )
      .then((response) => {
        if (
          response.data &&
          response.data.status &&
          response.data.status === "success"
        ) {
          setIsSnackbarOpen(true);
          setSnackBarSeverity("success");
          setSnackBarMessage("Transaction completed");
        }
      })
      .catch((error) => {
        if (error.response.data && error.response.data.status) {
          handleBackEndError(error);
        }
      })
      .finally(() => {
        setIsBackdropOpen(false);
        setSelectedRecipient("");
        setTransactionDate(new Date());
        setDescription("");
        setAmount("");
      });
  };

  const handleMakeTransferClick = () => {
    const errors = validateDetails();
    setErrorState(errors);
    if (Object.keys(errors).length !== 0) {
      setErrorState(errors);
      return;
    } else {
      setErrorState({});
    }
    makeTransfer();
  };

  return (
    <div className="makeTransferContainer">
      <CustomSnackBar
        open={isSnackbarOpen}
        severity={snackBarSeverity}
        message={snackBarMessage}
        handleClose={() => setIsSnackbarOpen(false)}
      />
      <CustomBackdrop
        open={isBackdropOpen}
        handleClose={() => setIsBackdropOpen(false)}
      />
      <NavBar />
      <div className="transferForm">
        <div style={{ fontSize: "25px", fontWeight: 700, color: "#dd54a5" }}>
          Make a transfer
        </div>
        <div className="formCenter">
          <div>
            <FormControl fullWidth error={errorState["selectedRecipient"]}>
              <InputLabel fullWidth id="demo-simple-select-helper-label">
                Select Recipient
              </InputLabel>
              <Select
                fullWidth
                labelId="demo-simple-select-helper-label"
                id="demo-simple-select-helper"
                value={selectedRecipient}
                label="Select Recipient"
                style={{ textAlign: "left" }}
                onChange={handleRecipientChange}
              >
                <MenuItem value="">
                  <em>None</em>
                </MenuItem>
                {recipient.map((data) => (
                  <MenuItem value={data.accountNo || ""}>
                    {data.accountHolderName || ""}
                  </MenuItem>
                ))}
              </Select>
            </FormControl>
          </div>
          <div className="formElement">
            <LocalizationProvider dateAdapter={AdapterDateFns}>
              <DesktopDatePicker
                label="Date of transfer"
                id="datePicker"
                inputFormat="MM/dd/yyyy"
                fullWidth
                value={transactionDate}
                onChange={handleDateChange}
                renderInput={(params) => <TextField fullWidth {...params} />}
              />
            </LocalizationProvider>
          </div>
          <div className="formElement">
            <TextField
              label="Description"
              fullWidth
              id="description"
              className="textField"
              error={errorState["description"]}
              variant="outlined"
              value={description}
              onChange={(event) => handleTextBoxChange("description", event)}
            />
          </div>
          <div className="formElement">
            <TextField
              label="Amount"
              id="amount"
              fullWidth
              type="number"
              className="textField"
              error={errorState["amount"]}
              variant="outlined"
              value={amount}
              onChange={(event) => handleTextBoxChange("amount", event)}
            />
          </div>
        </div>
      </div>
      <div className="footer">
        <div className="buttonHolder">
          <CustomSecondaryButton
            id="cancel"
            buttonText="Cancel"
            buttonClick={handleCancelClick}
            icon={<CloseIcon />}
          />
        </div>
        <div className="buttonHolder">
          <CustomButton
            id="submit"
            buttonText="Submit"
            buttonClick={handleMakeTransferClick}
            icon={<PublishIcon />}
          />
        </div>
      </div>
    </div>
  );
};

export default MakeTransfer;
