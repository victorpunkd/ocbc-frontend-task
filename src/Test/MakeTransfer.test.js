import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import MakeTransfer from "../Components/MakeTransfer/MakeTransfer";

configure({ adapter: new Adapter() });

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

describe("MakeTransfer", () => {
  it("snapshot renders", () => {
    const component = renderer.create(<MakeTransfer />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should simulate demo-simple-select-helper onChange event", () => {
    const wrapper = shallow(<MakeTransfer />);
    expect(
      wrapper.find("#demo-simple-select-helper").simulate("change", {
        target: {
          value: "test",
        },
      })
    );

    expect(wrapper.find("#demo-simple-select-helper").prop("value")).toBe(
      "test"
    );
  });

  it("Should simulate description onChange event", () => {
    const wrapper = shallow(<MakeTransfer />);
    expect(
      wrapper.find("#description").simulate("change", {
        target: {
          value: "test",
        },
      })
    );

    expect(wrapper.find("#description").prop("value")).toBe("test");
  });

  it("Should simulate amount onChange event", () => {
    const wrapper = shallow(<MakeTransfer />);
    expect(
      wrapper.find("#amount").simulate("change", {
        target: {
          value: "test",
        },
      })
    );

    expect(wrapper.find("#amount").prop("value")).toBe("test");
  });

  it("Should simulate Cancel onClick event", () => {
    const wrapper = shallow(<MakeTransfer />);

    wrapper.find("#cancel").props().buttonClick("");
    expect(mockedUsedNavigate).toHaveBeenCalled();
  });

  it("Should simulate Submit onClick event", () => {
    const wrapper = shallow(<MakeTransfer />);

    wrapper.find("#description").simulate("change", {
      target: {
        value: "test",
      },
    });

    wrapper.find("#amount").simulate("change", {
      target: {
        value: "test",
      },
    });

    wrapper.find("#demo-simple-select-helper").simulate("change", {
      target: {
        value: "test",
      },
    });

    wrapper.find("#submit").props().buttonClick("");
    expect(wrapper.find("#submit").prop("id")).toBe("submit");
  });
});
