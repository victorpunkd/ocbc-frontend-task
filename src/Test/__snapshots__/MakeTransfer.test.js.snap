// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeTransfer snapshot renders 1`] = `
<div
  className="makeTransferContainer"
>
  <div>
    <div
      aria-hidden={true}
      className="MuiBackdrop-root css-1ji7omv-MuiBackdrop-root"
      onClick={[Function]}
      style={
        Object {
          "opacity": 0,
          "visibility": "hidden",
        }
      }
    >
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="navBarContainer"
  >
    <div
      className="logoutButton"
      onClick={[Function]}
    >
      Logout
    </div>
  </div>
  <div
    className="transferForm"
  >
    <div
      style={
        Object {
          "color": "#dd54a5",
          "fontSize": "25px",
          "fontWeight": 700,
        }
      }
    >
      Make a transfer
    </div>
    <div
      className="formCenter"
    >
      <div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            fullWidth={true}
            id="demo-simple-select-helper-label"
          >
            Select Recipient
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiSelect-root css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "left",
              }
            }
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-helper-label demo-simple-select-helper"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-simple-select-helper"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-dhh0px"
              >
                <span>
                  Select Recipient
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="formElement"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            Date of transfer
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="mm/dd/yyyy"
              required={false}
              type="tel"
              value="11/09/2021"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is Nov 9, 2021"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-dhh0px"
              >
                <span>
                  Date of transfer
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="formElement"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root textField css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="description"
            id="description-label"
          >
            Description
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="description"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-dhh0px"
              >
                <span>
                  Description
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="formElement"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root textField css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="amount"
            id="amount-label"
          >
            Amount
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="amount"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-dhh0px"
              >
                <span>
                  Amount
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="buttonHolder"
    >
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButtonBase-root makeStyles-buttonStyle-1 css-xsqe0f-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Cancel
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="buttonHolder"
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root makeStyles-buttonStyle-2 css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="submit"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Submit
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PublishIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
